doc/stdenv/stdenv.chapter.md

```diff
@@ -309,13 +309,51 @@ The attribute can also contain a list, a script followed by arguments to be pass
 passthru.updateScript = [ ../../update.sh pname "--requested-release=unstable" ];
 ```
 
-The script will be run with the `UPDATE_NIX_NAME`, `UPDATE_NIX_PNAME`, `UPDATE_NIX_OLD_VERSION` and `UPDATE_NIX_ATTR_PATH` environment variables set respectively to the name, pname, old version and attribute path of the package it is supposed to update.
+Finally, the attribute can be an attribute set, which, in addition to the command, can optionally provide the canonical attribute path to use for the list of extra features the script supports.
```

Comment:
Don't use "finally" or other indication of ordering in prose (same as we set `1.` for all numbered list elements in markdown) - order may change, and then the prose will have to change.

```suggestion
The attribute can be an attribute set. In addition to the command, it can optionally provide the canonical attribute path to use for the list of extra features the script supports.
```

Other than that, I don't understand that sentence. As I read it right now, `pname` is supposed to be somehow used "in the list of extra features". But how? The example doesn't show. Why? This is not explained here or in the next section.

I'd suggest to rather list the attributes and note what exactly each one is for, and then make a full example that actually works. It would be OK to reference an existing update script (link to source so it can be easily found for inspection) or referring to an example from previous sections.